packages/bundlers/default/src/DefaultBundler.js

```diff
@@ -66,45 +64,53 @@ export default new Bundler({
             });
             bundleByType.set(bundle.type, bundle);
             bundleRoots.set(bundle, [asset]);
-            siblingBundlesByAsset.set(asset.id, siblingBundles);
+            siblingBundlesByAsset.set(asset.id, []);
             bundleGraph.addBundleToBundleGroup(bundle, bundleGroup);
           }
 
           return {
             bundleGroup,
             bundleByType,
-            siblingBundles,
             bundleGroupDependency: dependency,
             parentNode: node,
           };
         }
 
         invariant(context != null);
         invariant(context.parentNode.type === 'asset');
+        let parentAsset = context.parentNode.value;
         let bundleGroup = nullthrows(context.bundleGroup);
         let bundleGroupDependency = nullthrows(context.bundleGroupDependency);
         let bundleByType = nullthrows(context.bundleByType);
-        let siblingBundles = nullthrows(context.siblingBundles);
+        let siblingBundles = nullthrows(
+          siblingBundlesByAsset.get(parentAsset.id),
+        );
+        let allSameType = assets.every(a => a.type === parentAsset.type);
 
         for (let asset of assets) {
-          // If any sibling bundles were created for this asset or its subtree previously,
-          // add them all to the current bundle group as well. This fixes cases where two entries
-          // depend on a shared asset which has siblings. Due to DFS, the subtree of the shared
-          // asset is only processed once, meaning any sibling bundles created due to type changes
-          // would only be connected to the first bundle group. To work around this, we store a list
-          // of sibling bundles for each asset in the graph, and when we re-visit a shared asset, we
-          // connect them all to the current bundle group as well.
           let siblings = siblingBundlesByAsset.get(asset.id);
-          if (siblings) {
-            for (let bundle of siblings) {
-              bundleGraph.addBundleToBundleGroup(bundle, bundleGroup);
-            }
-          } else {
-            siblingBundlesByAsset.set(asset.id, siblingBundles);
-          }
 
-          let parentAsset = context.parentNode.value;
           if (parentAsset.type === asset.type) {
+            // If any sibling bundles were created for this asset or its subtree previously,
```

Comment:
should this be moved into the consequent of the first `if` below and updated to reflect the change?

Reply:
you mean move the comment?